src/libs/execution_time_estimator/execution_time_estimator.cpp

```diff
@@ -96,4 +194,115 @@ Skill::parse_args(const std::string &args)
 	}
 }
 
+std::map<std::string, Skill>
+ExecutionTimeEstimator::get_skills_from_config(const std::string &path) const
+{
+	const int                                     ID       = 0;
+	const int                                     PROPERTY = 1;
+	std::unique_ptr<Configuration::ValueIterator> it(config_->search(path.c_str()));
+	std::map<std::string, std::string>            skill_strings;
+	while (it->next()) {
+		std::vector<std::string> skill_property =
+		  str_split(std::string(it->path()).substr(path.size()));
+		if (skill_property.size() != 2) {
+			break;
+		}
+		if (skill_property[PROPERTY] == "args") {
+			skill_strings[skill_property[ID]] += str_join(it->get_strings(), ',');
+		} else if (skill_property[PROPERTY] == "name") {
+			skill_strings[skill_property[ID]] =
+			  it->get_string() + "{" + skill_strings[skill_property[ID]];
+		}
+	}
+	std::map<std::string, Skill> res;
+	for (const auto &skill_string : skill_strings) {
+		res.insert(std::make_pair(skill_string.first, Skill(skill_string.second + "}")));
+	}
+	return res;
+}
+
+template <typename T>
+T
+ExecutionTimeEstimator::get_property(const Property<T> &property) const
+{
+	if (active_whitelist_entry_ == whitelist_.end()) {
+		return property.get_default_value();
+		;
+	} else {
+		return property.get_property(active_whitelist_entry_->first);
+	}
+}
+
+template <typename T>
+ExecutionTimeEstimator::Property<T>::Property(fawkes::Configuration * config,
+                                              const std::string &     path,
+                                              const std::string &     property,
+                                              const std::optional<T> &default_val)
+{
+	try {
+		if constexpr (std::is_same<T, std::string>()) {
+			default_value = config->get_string((path + property).c_str());
+		} else if constexpr (std::is_same<T, float>()) {
+			default_value = config->get_float((path + property).c_str());
+		} else if constexpr (std::is_same<T, bool>()) {
+			default_value = config->get_bool((path + property).c_str());
+		} else {
+			throw Exception("Unsupported type");
```

Comment:
I chose a different route of `static_assert` because the conditions of `enable_if` would have to be specified in source and header file. My solution is implemented in 2d3b8ebd4160d7b1ad034f22f695ab22c8ecfe96